schema/defs.json

```diff
@@ -0,0 +1,115 @@
+{
+  "description": "Definitions used throughout the OpenContainer Specification",
+  "definitions": {
+    "int8": {
+      "type": "integer",
+      "minimum": -128,
+      "maximum": 127
+    },
+    "int16": {
+      "type": "integer",
+      "minimum": -32768,
+      "maximum": 32767
+    },
+    "int32": {
+      "type": "integer",
+      "minimum": -2147483648,
+      "maximum": 2147483647
+    },
+    "int64": {
+      "type": "integer",
+      "minimum": -9223372036854775808,
+      "maximum": 9223372036854775807
+    },
+    "uint8": {
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 255
+    },
+    "uint16": {
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 65535
+    },
+    "uint32": {
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 4294967295
+    },
+    "uint64": {
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 18446744073709551615
+    },
+    "uint16Pointer": {
+      "oneOf": [
+        { "$ref": "#/definitions/uint16" },
+        { "type": "null" }
+      ]
+    },
+    "uint64Pointer": {
+      "oneOf": [
+        { "$ref": "#/definitions/uint64" },
+        { "type": "null" }
+      ]
+    },
+    "stringPointer": {
+      "oneOf": [
+        { "type": "string" },
+        { "type": "null" }
+      ]
+    },
+    "mapStringString": {
+      "type": "object",
+      "patternProperties": {
+        ".{1,}": { "type": "string" }
+      }
+    },
+    "UID": { "$ref": "#/definitions/uint32" },
+    "GID": { "$ref": "#/definitions/uint32" },
+    "ArrayOfGIDs": {
+      "type": "array",
+      "items": {
+        "oneOf": [ { "$ref": "#/definitions/GID" } ]
+      }
```

Comment:
On Thu, Mar 03, 2016 at 12:27:46PM -0800, Vincent Batts wrote:
> > +      "items": {
> > +        "oneOf": [ { "$ref": "#/definitions/GID" } ]
> > +      }
> 
> it is a typed array. Otherwise it would allow `["foo", 1, false]`

I'm pretty sure that's what the $ref-only phrasing [1] means.

[1]: https://github.com/opencontainers/specs/pull/313/files#r50946692